interface/main/messages/messages.php

```diff
@@ -409,6 +426,35 @@
         $("#new_note").submit();
     }
 });
+ function gotoReport(doc_id,pname,pid,pubpid,str_dob){
+		EncounterDateArray=new Array;
+        CalendarCategoryArray=new Array;
+        EncounterIdArray=new Array;
+        Count = 0;
+	<?php
+		if(isset($enc_list) && sqlNumRows($enc_list) >0 ){
+			while($row = sqlFetchArray($enc_list)){
+	?>
+				        EncounterIdArray[Count]='<?php echo attr($row['encounter']); ?>';
+                                        EncounterDateArray[Count]='<?php echo attr(oeFormatShortDate(date("Y-m-d", strtotime($row['date'])))); ?>';
+                                        CalendarCategoryArray[Count]='<?php echo attr(xl_appt_category($row['pc_catname'])); ?>';
+                                        Count++;
+	<?php
+			}
+		}
+	?>
+	$.ajax({
+		type:'post',
+		url:'<?php echo $GLOBALS['webroot']."/library/ajax/set_session_ajax.php";?>',
+		data:{set_pid: pid},
+		async: false
+	});
```

Comment:
Is the /library/ajax/set_session_ajax.php script and ajax call needed? There are several places in OpenEMR that change to a new patient where this is not necessary.